src/Component/Facet/JsonFacetTrait.php

```diff
@@ -51,30 +60,32 @@ public function getDomainFilter()
      * @param string $query
      * @param array  $bind  Bind values for placeholders in the query string
      *
-     * @return self Provides fluent interface
+     * @return \Solarium\Core\Configurable
      */
-    public function setDomainFilterQuery(string $query, array $bind = null): FacetSetInterface
+    public function setDomainFilterQuery(string $query, array $bind = null): Configurable
```

Comment:
I know that our interfaces aren't perfect. But returning FacetSetInterface might be better here.